pkg/deploy/controller/lifecycle/execnewpod/controller.go

```diff
@@ -0,0 +1,95 @@
+package execnewpod
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+// ExecNewPodController keeps an ExecNewPodAction pod's status in sync with
+// its associated deployment.
+//
+// Use ExecNewPodControllerControllerFactory to create this controller.
+type ExecNewPodController struct {
+	// deploymentClient provides access to deployments.
+	deploymentClient deploymentClient
+}
+
+// Handle syncs the action pod's status with its associated deployment.
+func (c *ExecNewPodController) Handle(pod *kapi.Pod) error {
```

Comment:
Will talk more tomorrow, but I found a problem with my approach when trying to deal with the higher level status persistence that has me thinking about other ways to approach the feature in general. Any ideas welcome right now- this POC may be the one I end up tossing now that I have more of the problems in my head.